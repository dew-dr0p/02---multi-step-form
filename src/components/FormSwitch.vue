<script setup lang="ts">
const props = defineProps<{
    modelValue: boolean
}>()
const emits = defineEmits<{
    (e: 'update:modelValue', value: any): void
}>()
</script>

<template>
        <input type="checkbox" id="switch" true-value="yearly" false-value="monthly" :checked="modelValue" @change="emits('update:modelValue', (<HTMLInputElement>$event.target).checked)" class="opacity-0 w-0 h-0">
        <label for="switch" class="relative inline-block w-10 h-5 top-0 left-0 right-0 bottom-0 cursor-pointer rounded-3xl bg-MarineBlue before:absolute before:h-3 before:w-3 before:rounded-[50%] before:bg-White before:left-1 before:top-1 before:transition-all"></label>
</template>

<style scoped>
input:checked + label::before {
    left: 1.5rem;
}
</style>