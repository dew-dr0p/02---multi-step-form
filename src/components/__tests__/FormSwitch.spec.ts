import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import FormSwitch from "../FormSwitch.vue";

describe('FormSwitch', () => {
    const wrapper = mount(FormSwitch, {
        props: {
            modelValue: false,
            'onUpdate:modelValue': (e: boolean) => wrapper.setProps({modelValue: e})
        }
    })

    it('renders properly', () => {

        expect(wrapper.get('label')).toMatchSnapshot()
    })

    it('updates on click', async () => {

        await wrapper.find('input').setValue()

        expect(wrapper.props('modelValue')).toBe(true)
    })
})