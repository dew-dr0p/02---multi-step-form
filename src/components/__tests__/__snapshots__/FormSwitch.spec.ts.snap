// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormSwitch > renders properly 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <label
    class="relative inline-block w-10 h-5 top-0 left-0 right-0 bottom-0 cursor-pointer rounded-3xl bg-MarineBlue before:absolute before:h-3 before:w-3 before:rounded-[50%] before:bg-White before:left-1 before:top-1 before:transition-all"
    data-v-2566e8dc=""
    for="switch"
  />,
}
`;
